### DISCUSSION TITLE
Is there a way to get video capture source device id?

### DISCUSSION THREAD

**Discussion by rakusan2** [Palakis/obs-websocket#893] [Q&A] [answered]
I am trying to completely automate the setup of a scene for my own project and for that, I need to be able to add a webcam to the scene which requires a `video_device_id`
Is there a way for me to get this device id?

**Comment by tt2468** [comment, marked as answer]
In 5.x, use `GetInputPropertiesListPropertyItems` to fetch an array of dropdown items from the properties of the capture device source.

**Reply by igs-bingjiunchen** [reply]
sorry to bother, I'm not sure the parameter `PropertyName` should I provide, would you mind provided some example code??
